editor/src/templates/image-dnd.spec.browser2.tsx

```diff
@@ -353,7 +355,105 @@ export var storyboard = (
         width: 1,
         height: 1,
       }}
-      data-uid='imgimgimg'
+      data-uid='1'
+      data-aspect-ratio-locked
+    />
+  </Storyboard>
+)
+`)
+  })
+
+  it('dragging multiple images from the "finder" works', async () => {
+    FOR_TESTS_setNextGeneratedUids(['1', '2', '3'])
+
+    const editor = await renderTestEditorWithProjectContent(contents, 'await-first-dom-report')
+    const canvasControlsLayer = editor.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const files = [
+      await makeImageFile(imgBase64, 'chucknorris.png'),
+      await makeImageFile(imgBase64, 'budspencer.png'),
+      await makeImageFile(imgBase64, 'brucelee.png'),
+    ]
+
+    const target = editor.renderedDOM.getByTestId('scene')
+    const targetBounds = target.getBoundingClientRect()
+
+    const endPoint = {
+      x: targetBounds.x + targetBounds.width / 2,
+      y: targetBounds.y + targetBounds.height / 2,
+    }
+
+    fireEvent(
+      canvasControlsLayer,
+      makeDragEvent('drag', canvasControlsLayer, { x: 5, y: 5 }, files),
+    )
+
+    fireEvent(canvasControlsLayer, makeDragEvent('drag', canvasControlsLayer, endPoint, files))
+
+    fireEvent(canvasControlsLayer, makeDragEvent('drop', canvasControlsLayer, endPoint, files))
+
+    await editor.getDispatchFollowUpActionsFinished()
+
+    await wait(250) // read the image
```

Comment:
is there a way to actually await some sort of event that lets us know the image is ready?

Reply:
so the problem is that while action dispatch is synchronous, the `SAVE_ASSET` action fires off a bunch of async actions (such as reading image data and dimensions), and those are necessary for the action to really take effect. I'll check whether there's a "low hanging fruit" solution for this